### DISCUSSION TITLE
WSL2 - Hyptonix Application No audio

### DISCUSSION THREAD

**Discussion by vinodkrishnavarma** [microsoft/WSL#8331] [General]
### Version

Windows11 21H2

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.10.102.1

### Distro Version

20.04

### Other Software

Hypnotix 1.1

### Repro Steps

1. Install WSL2
2. Install Ubuntu 
3. run wget https://github.com/linuxmint/hypnotix/releases/download/1.1/hypnotix_1.1_all.deb
4. sudo dpkg -i hypnotix_1.1_all.deb
5. sudo apt -f install
6. hypnotix

### Expected Behavior

It should open hypnotix IPTV application and should able to select the tv channel.

### Actual Behavior

The hypnotix application will Open, but when we select any channel there is a GUI graphics problem. I can see the below errors in the ubuntu console
Could not load badge /usr/share/hypnotix/pictures/badges/usa.png
Traceback (most recent call last):
  File "/usr/lib/hypnotix/hypnotix.py", line 293, in on_category_button_clicked
    self.show_channels(group.channels)
  File "/usr/lib/hypnotix/hypnotix.py", line 323, in show_channels
    surface = Gdk.cairo_surface_create_from_pixbuf(pixbuf, self.window.get_scale_factor())
TypeError: Couldn't find foreign struct converter for 'cairo.Surface'

### Diagnostic Logs

Could not load badge /usr/share/hypnotix/pictures/badges/usa.png
Traceback (most recent call last):
  File "/usr/lib/hypnotix/hypnotix.py", line 293, in on_category_button_clicked
    self.show_channels(group.channels)
  File "/usr/lib/hypnotix/hypnotix.py", line 323, in show_channels
    surface = Gdk.cairo_surface_create_from_pixbuf(pixbuf, self.window.get_scale_factor())
TypeError: Couldn't find foreign struct converter for 'cairo.Surface'

**Comment by vinodkrishnavarma** [comment]
Application is working now after installing the below dep.
sudo apt install python3-gi-cairo

But now video is displaying correctly but no audio.

**Comment by ghost** [comment]
There is currently no native sound support for WSL, which is why you are not hearing any audio. It appears that the video issue was due to you lacking the dependency stated above- I will convert this to a discussion so that you can get tips from other WSL users on how to work around the sound issue
